### PR TITLE
Implement UI indicating that Internet connection is necessary to search new ARASAAC and Global Symbols

### LINKED ISSUE #1268: Implement UI indicating that Internet connection is nescesary to search new ARASAAC and Global Symbols

### PR DESCRIPTION
**Pull request by yehwang1**
Closes #1268 

**Issue description & Why is this change needed?**

- Currently, there is no description in the Cboard application notifying the user when the internet network is down, especially while searching through the symbol library. This notification is needed so that users know that global symbols and ARASAAC can only be searched for / retrieved while the user is connected to the internet since this would expand the search results available.

**What is the change and how we fixed the issue?** 
- The code change we added to fix the issue is that when the internet goes offline, a new message will appear on the Cboard symbol search page that explains how Internet is necessary to access global symbol and ARASAAC results.
To recreate this, a user can log on to the application, then click the lock 5 times to unlock the Add Tile button. After clicking the add tile button, the user can click the Symbols button to navigate to the list of symbols. By turning Wi-Fi on and off and attempting to search for global symbols and ARASAAC symbols, the user can verify that the UI message is shown.

**Notes / Screenshots of UI change**
- When the internet connection cuts, an interaction with the Cboard interface must occur for the react to re-render the react UI notification. This re-rendering will not occur instantaneously, but only after the user interacts with the search bar or filters. The instantaneous appearance of the message the moment the internet is cut should not be an issue for every case that we considered.

Before (internet online)
Screen Shot 2022-11-16 at 8 57 17 PM

After (internet offline)
Screen Shot 2022-11-16 at 8 57 29 PM

### CODE REVIEW

**Comment by yehwang1** [comment]
We are currently writing tests for this new feature and we are pushing this initial PR for review in the meantime. We plan to update this PR with tests within one week.

**Review by martinbedouret — CHANGES_REQUESTED** [review]
@yehwang1 Thanks for your fix! Some comments BELOW: 

- For the notifications to the users, please use the notification component we use throughout the entire application to be compliant with the material design principles https://m2.material.io/design/introduction
- The notification should be a warning type 
- The text message must be taken from a message file and included in the cboard.json for translations. 

@tomivm can you help with this review?

**Comment by tomivm** [comment]
Hi! it's a pleasure.
To show a Notification should pass the "showNotification" action from Board.container (Board.container.js:1667) through "TileEditor" and "SymbolSearch" components via props.
With the objective to show a 'warning' notification. Send the message on the first argument and a 'warning' string on the second argument. I will create that kind of notification for this PR
It should show the notification only after the 'SymbolSearch' component is mounted and the app is not connected to the internet.
The notification message should be similar to 'Available symbols are limited during editing offline'. please @martinbedouret review this message.
The message would be included on src/translations/src/cboard.json and on the src/components/Board/SymbolSearch/SymbolSearch.messages.js
Should use intl.formatMessage() with the message like an argument for `showNotification()` to display it. See this line as an example: https://github.com/cboard-org/cboard/blob/1ea47033f162aa8581a4bbd36e3c7b729e750d53/src/components/Analytics/Analytics.container.js#L89
Let me know if you need some help! you can contact me on discord tomivm#2865

**Comment by yehwang1** [comment]
Hi Martin and Tom, thank you for leaving such great feedback! My partner and I will address your suggestions and give an update in about 1.5 weeks from now. Thank you!

File: src/components/Board/TileEditor/TileEditor.component.js [hunk 1]
```diff
@@ -679,6 +686,7 @@ export class TileEditor extends Component {
             autoFill={this.state.autoFill}
             onChange={this.handleSymbolSearchChange}
             onClose={this.handleSymbolSearchClose}
+            intl={this.props.intl}
```
Comment (tomivm) [inline src/components/Board/TileEditor/TileEditor.component.js]: this prop is never used.

File: src/components/Board/SymbolSearch/SymbolSearch.css [hunk 2]
```diff
@@ -33,7 +33,7 @@
   left: 0;
   width: 100%;
   overflow: hidden;
-  z-index: 1;
+  /* z-index: 0; */
```
Comment (tomivm) [inline src/components/Board/SymbolSearch/SymbolSearch.css]: this change is necessary?

File: src/components/Board/SymbolSearch/SymbolSearch.css [hunk 3]
```diff
@@ -43,7 +43,8 @@
 
 @media (min-width: 600px) {
   .react-autosuggest__suggestions-container {
-    top: 128px;
+    /* Originally this was 128px */
+    top: 170px;
```
Comment (tomivm) [inline src/components/Board/SymbolSearch/SymbolSearch.css]: why this change?

File: src/components/Board/SymbolSearch/SymbolSearch.css [hunk 4]
```diff
@@ -84,3 +85,16 @@
 .SymbolSearch__Suggestion--highlighted {
   border-color: #9d3fb5;
 }
+
+#offlineMsg {
```
Comment (tomivm) [inline src/components/Board/SymbolSearch/SymbolSearch.css]: this is never used

File: src/components/UI/FilterBar/FilterBar.css [hunk 5]
```diff
@@ -1,6 +1,7 @@
 .FilterBar {
   display: flex;
-  margin-top: 1px;
+  margin-top: 10px;
```
Comment (tomivm) [inline src/components/UI/FilterBar/FilterBar.css]: this change is not necessary

**Review by tomivm — CHANGES_REQUESTED** [review]

File: src/components/Board/TileEditor/TileEditor.test.js [hunk 6]
```diff
@@ -72,4 +77,50 @@ describe('TileEditor tests', () => {
...
+    const wrapper = shallow(<TileEditor {...props} />);
+
+    // HIGH LEVEL LOGIC
+    // 1. We want to render tileEditor and SymbolSearch (thus use mount function to render children too)
+    // 2. We want to click on the button on tileEditor which will transition to symbolsearch component
+    // 3. Then check DOM for the showNotification message
+
+    console.log(wrapper.debug());
+
+    wrapper.find('.hello').simulate('click');
```
Comment (tomivm) [inline src/components/Board/TileEditor/TileEditor.test.js:91]: the new tests are failling 
```FAIL  src/components/Board/TileEditor/TileEditor.test.js (46.4s)
  ● TileEditor tests › should render notification if offline

    Method “simulate” is meant to be run on 1 node. 0 found instead.

      91 |     console.log(wrapper.debug());
      92 |
    > 93 |     wrapper.find('.hello').simulate('click');
         |                            ^
      94 |     console.log('BUTTON CLICKED');
      95 |     console.log(wrapper.debug());
      96 |     // expect(onButtonClick).to.have.property('callCount', 1);

      at ShallowWrapper.single (node_modules/enzyme/src/ShallowWrapper.js:1652:13)
      at ShallowWrapper.simulate (node_modules/enzyme/src/ShallowWrapper.js:1133:17)
      at Object.<anonymous> (src/components/Board/TileEditor/TileEditor.test.js:93:28)
```

File: src/components/Board/TileEditor/TileEditor.test.js [hunk 7]
```diff
@@ -72,4 +77,50 @@ describe('TileEditor tests', () => {
...
+    console.log('BUTTON CLICKED');
+    console.log(wrapper.debug());
```
Comment (tomivm) [inline src/components/Board/TileEditor/TileEditor.test.js]: remove console.log 's

**Review by tomivm — CHANGES_REQUESTED** [review]

**Comment by martinbedouret** [comment]
closed due to inactivity

**Pull request closed without merging**